docs/source/library-user-guide/api-health.md

```diff
@@ -0,0 +1,37 @@
+<!---
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.
+-->
+
+# API health policy
```

Comment:
I think we should also add a link to the main README as well  -- perhaps something on https://github.com/apache/datafusion?tab=readme-ov-file#apache-datafusion?

Reply:
Sorry @alamb i'm not quite following

Reply:
I was thinking that since this policy is in its own file (not in the README) it is not visible to people when they open https://github.com/apache/datafusion

I think it would make it easier to find if we either put the policy directly in the https://github.com/apache/datafusion/blob/main/README.md file or if not left a link in that file that points here

Reply:
That totally makes sense

Reply:
Added small policy description and a link from README to the policy